cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -454,6 +483,12 @@ public Cancellable announce(String serviceName, int port) {
 
           // For binding StreamWriter
           install(createStreamFactoryModule());
+
+          // bind AuthorizationEnforcer to AuthorizationEnforcementService
+          bind(AuthorizationEnforcer.class).to(AuthorizationEnforcementService.class).in(Scopes.SINGLETON);
+          // also bind AuthorizationEnforcementService as a singleton. This binding is used while starting/stopping
+          // the service itself.
+          bind(AuthorizationEnforcementService.class).in(Scopes.SINGLETON);
```

Comment:
``AuthorizationEnforcementService`` has two roles:
1. An ``AbstractScheduledService`` that refreshes privileges periodically. A handle to this is obtained using ``injector.getInstance(AuthorizationEnforcementService.class)``.
2. An ``AuthorizationEnforcer``, which is used by the dataset methods enforce authorization. Presumably, a handle to this will be obtained using ``injector.getInstance(AuthorizationEnforcer.class)``.

Is there a better way to do this?

Also, I didn't want to make ``AuthorizationEnforcementService`` an ``Authorizer``, because ``Authorizer`` can modify privileges as well, and I didn't want to expose that functionality in ``AuthorizationEnforcementService``.